packages/web/src/pages/upload-page/UploadPageNew.tsx

```diff
@@ -69,21 +73,41 @@ export const UploadPageNew = () => {
     injectPrettifyScript()
   }, [phase])
 
+  useEffect(() => {
+    // Reset if user removes all tracks during collection upload
+    if (phase !== Phase.SELECT && tracks?.length === 0) {
+      setFormState({
+        uploadType: undefined,
+        metadata: undefined,
+        tracks: undefined
+      })
+      setPhase(Phase.SELECT)
+    }
+  }, [phase, tracks, uploadType])
+
   const pageTitle = useMemo(() => {
     switch (phase) {
       case Phase.EDIT:
-        return tracks && tracks.length > 1
+        return uploadType === UploadType.ALBUM
+          ? messages.editAlbumPageTitle
+          : uploadType === UploadType.PLAYLIST
+          ? messages.editPlaylistPageTitle
+          : tracks && tracks.length > 1
```

Comment:
this block is getting hard to read. Can we factor our some variables or something?

Reply:
yea good thought